libethcore/KeyManager.cpp

```diff
@@ -422,12 +422,20 @@ void KeyManager::write(SecureFixedHash<16> const& _key, string const& _keysFile)
 {
 	RLPStream s(4);
 	s << 1; // version
-	s.appendList(accounts().size());
+
+	s.appendList(m_keyInfo.size());
 	for (auto const& address: accounts())
 	{
-		h128 id = uuid(address);
-		auto const& ki = m_keyInfo.at(address);
-		s.appendList(5) << address << id << ki.passHash << ki.accountName << ki.passwordHint;
+		try
```

Comment:
Don't use exceptions in C++ if you can avoid them, they are really expensive.
So please rather use something like: `if (!m_keyinfo.count(address)) continue;`